### PR TITLE
Tab close buttons

### PR DESCRIPTION
**Pull request by mkenne11**
This change adds close buttons to tabs - these are visible when more than one tab is open. A sample of the close buttons is shown in the following screenshot.

image

The close button changes on focus. See additional sample screenshot below.

image

### CODE REVIEW

**Comment by mkenne11** [comment]
@shockone I am not sure what you had in mind when it came to mouse/ui design for tabs. Please let me if this is close to something you expect or not. I am happy to make changes to get closer to what you wanted.

File: src/views/1_ApplicationComponent.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export default class ApplicationComponent extends React.Component<{}, State> {
             }
         }
 
-        if (event.metaKey && event.keyCode === KeyCode.T) {
+        if ((event.metaKey || event.ctrlKey) && event.keyCode === KeyCode.T) {
```
Comment (shockone) [inline src/views/1_ApplicationComponent.tsx]: This is not related to the subject. Please remove this change.

File: src/views/TabComponent.tsx [hunk 2]
```diff
@@ -8,12 +8,14 @@ export interface TabProps {
     isActive: boolean;
     activate: () => void;
     position: number;
+    closetab: () => void;
```
Comment (shockone) [inline src/views/TabComponent.tsx]: Should be `closeHandler`.

File: src/views/1_ApplicationComponent.tsx [hunk 3]
```diff
@@ -140,6 +141,20 @@ export default class ApplicationComponent extends React.Component<{}, State> {
...
+        let selectedTabIndex: number = Number(event.target.value);
+
+        if (selectedTabIndex === this.activeTabIndex) {
+            this.closeSession(this.activeTab.activeSession());
+        } else {
+            this.closeTab(this.tabs[selectedTabIndex]);
+            if (selectedTabIndex < this.activeTabIndex)
+                this.activeTabIndex -= 1;
+            this.setState({sessions: this.activeTab.sessions});
+        }
```
Comment (shockone) [inline src/views/1_ApplicationComponent.tsx]: It really should be the same as `event.metaKey && event.keyCode === KeyCode.W` handling (except selecting the  correct tab to close).

**Comment by shockone** [comment]
Thank you for the contribution, @mkenne11. Design-wise I'd like the close button to appear on the left side (as in Safari tabs), and also it would be cool it it appeared only on hover to have less visual noise.

**Comment by mkenne11** [comment]
@shockone I believe I made the changes you requested. Please let me know if any extra changes are required :)

**Comment by shockone** [comment]
Thanks, @mkenne11. I'll merge it. I think I know a better way to handle closing, but I can modify it by myself instead of bothering you. Also, it looks buggy that the tab number is not aligned properly, but we can fix that separately. 

Two more things:
* Please don't use color values, try to get them from the map. You'll find a lot of examples in the CSS file. It will help when we modify the theme or implement themes support.
* How strong do you feel about having the top line? I thought that distinguishing tabs by color should be enough. Besides, it makes the number move down, which also looks buggy.

**Comment by mkenne11** [comment]
No worries @shockone. I can see your point about using colour values from the map - you have a standard colour palette and you'd like like to use that - definitely makes things more consistent.

I don't feel that strongly about having the top border, it was an idea I had but I'm not wedded to it. And can definetly see your point that moving the number down makes it look buggy.

**Comment by shockone** [comment]
@mkenne11, I've eliminated the need for the `value` attribute by binding the index to the handler as we iterate over them. https://github.com/shockone/black-screen/commit/fcd7184714e95eae1170b5286a5489fdd4166a39

**Comment by mkenne11** [comment]
Thanks @shockone that looks cleaner.

**Comment by shockone** [comment]
I've messed with CSS quite a bit. Now it looks closer to the original tabs design. https://github.com/shockone/black-screen/commit/23921e449266d942a146aae628fa1cf9c7569f40

On hover:

On icon hover:


Hopefully I haven't offended you by modifying your work.

**Comment by mkenne11** [comment]
No probs at all I'm not offended :) your the main author of the project and have a vision for how things should look and feel. Plus I'd expect you'd want the "cleanest" commits possible to minimise code maintenance over time.

**Comment by shockone** [comment]
Awesome, @mkenne11. I'm glad you're cool with it.
